### PR TITLE
Address Rust 1.49 issues

### PR DESCRIPTION
**Pull request by rbradford**
The new toolchain introduced some new clippy checks as well as general build checks.

### CODE REVIEW

File: arch/src/x86_64/smbios.rs [hunk 1]
```diff
@@ -157,7 +157,7 @@ fn write_string(
     for c in val.as_bytes().iter() {
         curptr = write_and_incr(mem, *c, curptr)?;
     }
-    curptr = write_and_incr(mem, 0 as u8, curptr)?;
+    curptr = write_and_incr(mem, 0u8, curptr)?;
```
Comment (bryteise) [inline arch/src/x86_64/smbios.rs:160]: Is there a reason you didn't have this type of change the avoid cast PR?

File: hypervisor/src/kvm/mod.rs [hunk 2]
```diff
@@ -1234,12 +1234,14 @@ impl cpu::Vcpu for KvmVcpu {
     ///
     #[cfg(target_arch = "aarch64")]
     fn state(&self) -> cpu::Result<CpuState> {
-        let mut state = CpuState::default();
+        let mut state = CpuState {
+            mp_state: self.get_mp_state()?,
+            mpidr: self.read_mpidr()?,
+            ..Default::default()
+        };
         // Get this vCPUs multiprocessing state.
```
Comment (bryteise) [inline hypervisor/src/kvm/mod.rs:1242]: This comment should be moved or removed now.

**Comment by bryteise** [comment]
Gotta love new clippy advancements. Those unnecessary casts were nice and contains was new to me too! Also shout out to github's beta "Unchanged files with check annotations" telling me what the next commit was going to be lol.

File: arch/src/x86_64/smbios.rs (hunk 1 above)
Comment (rbradford) [reply arch/src/x86_64/smbios.rs:160]: We need to keep the specific type here a the function takes a T and we need it to be 1 byte only.

Comment (rbradford) [reply arch/src/x86_64/smbios.rs:160]: You're right though, this shouldn't be in this commit.

**Review by sboeuf — APPROVED** [review]

**Pull request merged**